src/github3/api.py

```diff
@@ -34,6 +39,10 @@ def authorize(username, password, scopes, note='', note_url='', client_id='',
     :returns: :class:`Authorization <Authorization>`
 
     """
+    warnings.warn(
+        'The anonymous functional API is deprecated. Use the `GitHub` object '
```

Comment:
If the message isn't going to be customized for each function, we could abstract either the string or the call to a private function.

Do you think it's worth doing:

```py
def _deprecated(func_name):
    warnings.warn(
        'The anonymous API function `github3.api.{0}` is deprecated. Use `GitHub.{0}` instead.'.format(func_name),
        DeprecationWarning
    )
```

And then call it like:

```py
    _deprecated('authorize')
    # ...
    _deprecated('emojis')
    # etc.
```

This might help users understand *what* they're doing that's deprecated

Reply:
I like the idea of customizing the deprecation message per function. WDYT of the decorator-based design that I just pushed?